src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -413,7 +409,11 @@ public SqlCompareOptions CompareInfo
                 {
                     throw ADP.ArgumentOutOfRange(nameof(CompareInfo));
                 }
-                collation.SqlCompareOptions = value;
+
+                if (collation == null || collation.SqlCompareOptions != value)
+                {
+                    _collation = SqlCollation.FromLCIDAndSort(collation?.LCID ?? 0, value);
+                }
```

Comment:
how would this behave in async/concurrent calls? If I am not wrong, a new member of `SqlParameter` would be created not `SqlCollation` and we just do the clean up by using `FromLCIDAndSort` for sync calls. Am I correct?

Reply:
> Does that answer? I admit I don't fully understand the question.

Thanks it did answer my question. Apparently I missed a part of the code and I got myself confused as well.